File: web_version/pmugg_files/util/counter.js
ms.counter = {};
ms.counter.props = {};

ms.counter.register = function(name) {
	ms.counter.props[name] = 0;
};

ms.counter.add = function(name) {
	return ms.counter.props[name]++;
}

ms.counter.reset = function() {
	Object.keys(ms.counter.props).forEach((key) => {
		ms.counter.props[key] = 0;
	});
};

File: cpp_version/pmugg/util/counter.cpp
#include "pch.h"
#include "counter.h"

namespace ms {

void Counter::register_(const std::string& name) {
    props[name] = 0;
}

int Counter::add(const std::string& name) {
    return ++props[name];
}

void Counter::reset() {
    for (auto& [key, value] : props) {
        value = 0;
    }
}

} // namespace ms 